ansible/roles/cnf_profile/templates/performance-profile.yml.j2

```diff
@@ -0,0 +1,33 @@
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+# This profile is for typical lab DELL R640 with 2 x 26 pcores
+  name: perf-example
+spec:
+  additionalKernelArgs:
+  - nosmt
+  cpu:
+    # Temp. workaround for RT kernel bugs that consume too much 
+    # CPU power: add more isolated cores
+    isolated: "1,3,5,7,9-51"
+    reserved: "0,2,4,6,8"
+  hugepages:
+    defaultHugepagesSize: 1G
+    pages:
+    - count: {{ hugepage_count }}
+      size: {{ hugepage_size }}
+      node: {{ hugepage_node }}
+  # for 3 node converged master/worker and SNO clusters we use the masters as a selector
+  nodeSelector:
+    node-role.kubernetes.io/master: ""
+  machineConfigPoolSelector:
+    pools.operator.machineconfiguration.openshift.io/master: ""
```

Comment:
Not sure I follow here - my understanding is that a PerformanceProfile is applied to a MCP and its associated hosts - what am I missing?

Reply:
Yes you are right, also this will change based to the `nodeSelector` label. 
For example,
```
   nodeSelector:
     node-role.kubernetes.io/worker-cnf: ""
   machineConfigPoolSelector:
     machineconfiguration.openshift.io/worker-cnf: ""
```

Reply:
I think what @mukrishn means is, as per performance-add-on oprator docs, the machineconfigpoolselector is optional, 

```MachineConfigPoolSelector defines the MachineConfigPool label to use in the MachineConfigPoolSelector of resources like KubeletConfigs created by the operator. Defaults to "machineconfiguration.openshift.io/role=<same role as in NodeSelector label key>"
```
If we were to only use Nodeselector within the performanceprofile, then we need add this `machineconfiguration.openshift.io/role=master` label to the MCP, so I'm Ok with just leaving things the way they are. The bigger question is, I think we need to take as input from user the nodes to apply this profile on, as we are assuming its always the masters here. I think we should variabilize that part.

Reply:
I've turned the selectors, or at least the variable part of it, into templates. Please have a look and let me know if you'd do it differently, I didn't find a really nice way of doing it.